types/react-transition-group/SwitchTransition.d.ts

```diff
@@ -0,0 +1,51 @@
+import { Component } from "react";
+
+export enum modes {
+  out = 'out-in',
+  in = 'in-out'
+}
+
+declare namespace SwitchTransition {
+  interface SwitchTransitionProps {
+    /**
+     * Transition modes.
+     * `out-in`: Current element transitions out first, then when complete, the new element transitions in.
+     * `in-out: New element transitions in first, then when complete, the current element transitions out.`
+     */
+    mode?: 'out-in' | 'in-out';
```

Comment:
It seems possible to use `enum modes` here instead of string literals. What about?

```suggestion
    mode?: modes
```

Reply:
Hey there 👋 

I did try this, but TypeScript is pretty strict when I change it to this, and will require that you always use the `enum modes` export when you pass in the React prop. 

So the following example would be a type error:
```tsx
import SwitchTransition from 'react-transition-group/SwitchTransition';

// results in type error
<SwitchTransition mode="in-out">
  {...}
</SwitchTransition>
```

Whereas this would not result in an error:
```tsx
import SwitchTransition, { modes } from 'react-transition-group/SwitchTransition';

// we're using the enum directly, so this is 👌🏼
<SwitchTransition mode={modes.in}>
  {...}
</SwitchTransition>
```

Reply:
I see👍

Reply:
If I can weigh in on this, I think it's much better to allow the string ( `<SwitchTransition mode="in-out">` ), since it will 

* be more in line with the official docs
* be easier to type (no need to figure out the import)
* have the same type safety (invalid values still not allowed)

Reply:
Agreed. I've left this as is, so the current state of this PR implements the behavior you're describing 👍

Reply:
Ah, ok, good to hear!